### PR TITLE
chore: use k8s 1.27 for CI

### PR DESCRIPTION
**Pull request by tico24**
### Motivation

We currently test against k8s 1.21. This has been out of support for some time (since 2022-06-28), nobody should be using it.
This PR updates the CI to run against k8s 1.27. 1.27 goes EOL on 2024-06-28.

### Modifications
- Updated CI to test against k8s 1.27
- Modifed docs to explain this fact.

### Verification
CI passes. (it has passed, there's some github issue causing the results to not come back to prs properly)

### CODE REVIEW

**Review by JPZ13 — APPROVED** [review]
Glad the upgrade is working on the CI. We're running Argo WF on K8s 1.27, so I'll approve

File: .github/workflows/ci-build.yaml [hunk 1]
```diff
@@ -87,13 +87,13 @@ jobs:
           - test: test-python-sdk
             profile: minimal
           - test: test-executor
-            install_k3s_version: v1.21.2+k3s1
```
[the remark this answers is not in the document]
Comment (terrytangyuan) [reply .github/workflows/ci-build.yaml:90]: Here is an example of using 1.26 https://github.com/argoproj/argo-workflows/actions/runs/5370677857/jobs/9742961007

[the remark this answers is not in the document]
Comment (terrytangyuan) [reply .github/workflows/ci-build.yaml:90]: I think 1.25 makes sense. https://endoflife.date/kubernetes

**Review by terrytangyuan — APPROVED** [review]

**Comment by tico24** [comment]
oh. I just failed to type the right number! 1 sec...

**Pull request merged**